src/encoded/static/components/cart/cart.js

```diff
@@ -422,11 +423,12 @@ const CartTools = ({
     visualizable,
     isFileViewOnly,
 }) => {
-    // Make a list of all the dataset types currently in the cart.
-    const usedDatasetTypes = elements.reduce((types, element) => {
-        const type = atIdToType(element['@id']);
-        return types.includes(type) ? types : types.concat(type);
-    }, []);
+    // Make a list of all the allowed dataset types currently in the cart.
+    const usedDatasetTypes = new Set(
```

Comment:
I didn’t time a comparison, but I didn’t want to filter first because I’d have to search `allowedDatasetTypes` to find the matching type. By using map first, I can just do a direct lookup of `allowedDatasetTypes`.